ci-operator/step-registry/baremetalds/devscripts/gather/baremetalds-devscripts-gather-commands.sh

```diff
@@ -67,7 +70,28 @@ then
   . network.sh
   . utils.sh
 
-  ssh -o 'ConnectTimeout=5' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -o 'ServerAliveInterval=90' core@\$BOOTSTRAP_PROVISIONING_IP TAR_FILE=/tmp/log-bundle-bootstrap.tar.gz sudo -E /usr/local/bin/installer-gather.sh --id bootstrap &&
+  # Pass master IPs to installer-gather script to collect info from nodes which didn't join the cluster
+  NODE_IPS=()
+  for (( n=0; n<\$NUM_MASTERS; n++ ))
+  do
+     node_name=\$(printf \$MASTER_HOSTNAME_FORMAT \$n)
+     node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//dns[*]/host/hostname[. = '\${node_name}']/../@ip)" -)
+     NODE_IPS+=("\$node_ip")
+  done
+  for (( n=0; n<\$NUM_WORKERS; n++ ))
+  do
+     node_name=\$(printf \$WORKER_HOSTNAME_FORMAT \$n)
+     node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//dns[*]/host/hostname[. = '\$node_name']/../@ip)" -)
+     NODE_IPS+=("\$node_ip")
+  done
+  for (( n=0; n<\$NUM_EXTRA_WORKERS; n++ ))
+  do
+     node_name=\$(printf \$WORKER_HOSTNAME_FORMAT \$n)
+     node_ip=\$(sudo virsh net-dumpxml \$BAREMETAL_NETWORK_NAME | xmllint --xpath "string(//dns[*]/host/hostname[. = '\$node_name']/../@ip)" -)
+     NODE_IPS+=("\$node_ip")
+  done
+
+  ssh -o 'ConnectTimeout=5' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -o 'ServerAliveInterval=90' core@\$BOOTSTRAP_PROVISIONING_IP TAR_FILE=/tmp/log-bundle-bootstrap.tar.gz sudo -E /usr/local/bin/installer-gather.sh --id bootstrap \${NODE_IPS[@]} &&
```

Comment:
```ssh: connect to host 192.168.111.7 port 22: No route to host```
([from](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/34664/rehearse-34664-periodic-ci-openshift-assisted-service-master-edge-e2e-ai-operator-ztp-ipv4v6-sno-periodic/1634832751972585472/artifacts/e2e-ai-operator-ztp-ipv4v6-sno-periodic/baremetalds-devscripts-gather/build-log.txt) latest job)

iiuc it's only good if bootstrap node is kept alive, which might be problematic if we want to fetch spoke's logs

Thinking out loud, it's either keeping bootstrap alive, or maybe delegate this task to another component (a master node? or maybe copy the script to the provisioning host and run it from there)

Reply:
If bootstrap is not around there's no point to collect log bundle, instead we'll have must-gather. I can extend must-gather w/ logs from unprovisioned agents in a separate PR